.github/workflows/sdks.yml

```diff
@@ -146,28 +213,104 @@ jobs:
+            cat ${ARCH_JSON}
+          elif [[ ${{ matrix.os }} = 'macos'* ]]; then
+            # fix the clang symlink
+            ls -la ${SDK_PATH}/usr/lib/swift/clang
+            rm -vf ${SDK_PATH}/usr/lib/swift/clang
+            # Swift 5.10: usr/lib/clang/15.0.0 Trunk and Devel: usr/lib/clang/17
+            ${TOOLCHAIN}/bin/clang --version
+            ln -sfv ${TOOLCHAIN}/lib/clang/* ${SDK_PATH}/usr/lib/swift/clang
+            ls -d ${SDK_PATH}/usr/lib/swift/clang/include
+
+            # the toolchain destination JSON in the SDK is for ubuntu,
+            # so for macOS we need to manually build it
+            ARCH=${{ matrix.arch }}
+            ARCH_JSON="$PWD/android-${ARCH}.json"
+            ARCH_TARGET="${ARCH}-unknown-linux-android24"
+            ARCHID="${ARCH}-linux-android"
+            if [ "$ARCH" == "armv7" ]; then
+                ARCHID="arm-linux-androideabi"
+            fi
+
+            # current default on macos is 26.3.11579264,
+            # so use the path from the nttld/setup-ndk action for r27
+            ANDROID_NDK=${{ steps.setup-ndk.outputs.ndk-path }}
+            NDK_PREBUILT=${ANDROID_NDK}/toolchains/llvm/prebuilt/darwin-x86_64
+
+            # With the full SDK, the libs are all located in:
+            #"-L", "${SDK}/usr/lib/${ARCHID}"
+            # But with the partial SDK, they are divided between:
+            #"-L", "${SDK_PATH}/usr/lib",
+            #"-L", "${SDK_PATH}/usr/lib/swift/android",
+            cat > ${ARCH_JSON} << EOF
+            {
+                "version": 1,
+                "target": "${ARCH_TARGET}",
+                "toolchain-bin-dir": "${TOOLCHAIN}/bin",
+                "sdk": "${NDK_PREBUILT}/sysroot",
+                "extra-cc-flags": [ "-fPIC" ],
+                "extra-cpp-flags": [ "-lstdc++" ],
+                "extra-swiftc-flags": [
+                    "-resource-dir", "${SDK_PATH}/usr/lib/swift",
+                    "-tools-directory", "${NDK_PREBUILT}/bin",
+                    "-L", "${SDK_PATH}/usr/lib",
+                    "-L", "${SDK_PATH}/usr/lib/swift/android"
+                ]
+            }
+          EOF
+
+            cat ${ARCH_JSON}
 
-          if [ ! -d ${{ steps.version.outputs.tag }}-ubuntu22.04 ]; then
-            tar xf ~/${{ steps.version.outputs.tag }}-ubuntu22.04.tar.gz
+            echo "toolchain-bin: ${TOOLCHAIN}/bin"
+            ls -d ${TOOLCHAIN}/bin
+            echo "ANDROID_NDK: ${ANDROID_NDK}"
+            ls -d ${ANDROID_NDK}
+            echo "NDK_PREBUILT: ${NDK_PREBUILT}"
+            ls -d ${NDK_PREBUILT}/sysroot
+            echo "resource-dir: ${SDK_PATH}/usr/lib/swift"
+            ls -d ${SDK_PATH}/usr/lib/swift
+            echo "tools-directory: ${NDK_PREBUILT}/bin"
+            ls -d ${NDK_PREBUILT}/bin
+            echo "-L: ${SDK_PATH}/usr/lib"
+            ls -d ${SDK_PATH}/usr/lib
+            echo "-L: ${SDK_PATH}/usr/lib/swift/android"
+            ls -d ${SDK_PATH}/usr/lib/swift/android
```

Comment:
I love how verbose this pull is so far, but these last 14 lines should not be necessary anymore, since they're all dumped in the json file above anyway.